packages/natds-themes/config/shared/registerTypeDefinitionsAction.test.js

```diff
@@ -0,0 +1,96 @@
+import fs from 'fs';
+import { registerTypeDefinitionsAction } from './registerTypeDefinitionsAction';
+import * as templateHelpers from './templateHelpers';
+import { brands } from './constants';
+
+jest.mock('fs');
+
+const config = {
+  buildPath: 'build/web/avon',
+  files: [
+    { destination: 'light.json' },
+  ],
+};
+
+const tokens = {
+  borderRadius: {
+    none: 0,
+  },
+  color: {
+    primary: '#000000',
+  },
+};
+
+const expectedTypes = `export interface BorderRadius {
+\tnone: number
+}
+
+export interface Color {
+\tprimary: string
+}
+
+export interface Theme {
+\tborderRadius: BorderRadius
+\tcolor: Color
+}`;
+
+describe('registerTypeDefinitionsAction', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('do action', () => {
+    let typesTemplateSpy;
+    let compileTemplateSpy;
+
+    beforeEach(() => {
+      fs.readFileSync.mockImplementation(() => JSON.stringify(tokens));
+
+      typesTemplateSpy = jest
+        .fn()
+        .mockReturnValue('test');
+
+      compileTemplateSpy = jest
+        .spyOn(templateHelpers, 'compileTemplate')
+        .mockReturnValue(typesTemplateSpy);
+    });
+
+    it('should return false if definitions already exist', () => {
+      fs.existsSync.mockImplementation(() => true);
+      expect(registerTypeDefinitionsAction().do(null, config)).toEqual(false);
+    });
+
+    describe('creating a file definition', () => {
```

Comment:
Uma sugestão boba, não-bloqueante: em `describe`, eu recomendaria usar algo como `Type definition file` e nos `it` usar `should be created if there is none`. Describe ficaria mais como o item sendo testado, e o `it` funcionando praticamente como um título de critério de aceitação :)